File: Assignment9/program5.c
////////////////////////////////////////////////////////////////
//
//  File Name   : program5.c
//  Description : Accept a Integer and counts frequency of digits less than 6
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 17/05/2025
//
////////////////////////////////////////////////////////////////

#include<stdio.h>

////////////////////////////////////////////////////////////////
// 
//  Function Name : Count
//  Description   : Returns count frequency of digits less than 6 for input integer
//  Input         : Integer
//  Output        : Integer
//
////////////////////////////////////////////////////////////////

int Count(int iNo)
{
    int iDigit = 0;
    int iFreq = 0;
    
    if(iNo < 0)
    {
        iNo = -iNo;
    }
    
    while (iNo != 0)
    {
        iDigit=iNo%10;
        
        if(iDigit < 6)
        {
            iFreq  = iFreq + 1;
        }
        iNo = iNo / 10;
    }

    return iFreq ;
}

int main()
{
    int iValue = 0;
    int iRet = 0;
    
    printf("enter number");
    scanf("%d",&iValue);

    iRet= Count(iValue);
    
    printf("%d",iRet);
    
    return 0;
}

File: Assignment25/program5.java
////////////////////////////////////////////////////////////////
//
//  File Name   : program5.java
//  Description : Accept a character character array coverting it into string and returns Count of WhiteSpaces of that String
//  Author      : Akhilesh.P.Sonavane.
//  Date        : 29/05/2025
//
////////////////////////////////////////////////////////////////

import java.util.Scanner;

////////////////////////////////////////////////////////////////
// 
//  Class Name    : Display
//  Function Name : CountWhite
//  Description   : Inputs character array coverting it into string and and returns Count of WhiteSpaces of that String
//  Input         : String
//  Output        : Integer
//
////////////////////////////////////////////////////////////////

class Display
{
    public int CountWhite(char[] chVal, int iSize)
    {
        int iCnt = 0;
        int iCount = 0;

        for(iCnt = 0; iCnt < iSize; iCnt++)
        {
            if(chVal[iCnt] == ' ')
            {
                iCount++;
            }
        }
        return iCount;
    }
}

public class program5
{
    public static void main(String args[])
    {
        int iCnt = 0;
        char[] cValue = null;
        String sValue = null;
        int iLength = 0;
        int iRet = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter String : ");
        sValue = sobj.nextLine();

        iLength = sValue.length();

        cValue = new char[iLength];

        for(iCnt = 0; iCnt < iLength; iCnt++)
        {
            cValue[iCnt] = sValue.charAt(iCnt);
        }

        Display dobj = new Display();

        iRet = dobj.CountWhite(cValue, iLength);

        System.out.println("White Spaces in Entered String are : "+iRet);

        sobj.close();
    }
}
